src/PubSubClient.cpp

```diff
@@ -529,15 +529,14 @@ void PubSubClient::disconnect() {
 }
 
 uint16_t PubSubClient::writeString(const char* string, uint8_t* buf, uint16_t pos) {
-    const char* idp = string;
-    uint16_t i = 0;
-    pos += 2;
-    while (*idp) {
```

Comment:
Wasn't it enough just changing this line into `while (*idp && pos < (MQTT_MAX_PACKET_SIZE - 2)) {`   ??
Your code is much easier to read, but it must read the `string` twice while in the original implementation it was only read once.

Reply:
Agreed @TD-er - yours is the more efficient fix here. I'm going to manually apply it.